feat(adt): add content about Splunk integration MTA-7358 (#6744)
scaleway/docs-content: pages/audit-trail/reference-content/splunk-integration.mdx

categories:
  - monitoring
---

import Requirements from '@macros/iam/requirements.mdx'

import image1 from './assets/scaleway-audit-trail-splunk-add-account.png'

import image2 from './assets/scaleway-audit-trail-splunk-create-new-input.png'

Audit Trail supports sending event data to Splunk Enterprise and Splunk Cloud Platform for security monitoring, data visualization and analytics, and compliance reporting.

By centralizing Scaleway audit data within Splunk, security operations center (SOC) and cloud operations teams can address several operational and security use cases:
- User activity monitoring and access governance
- Change tracking and configuration management
- Compliance auditing and reporting (SOC 2, ISO 27001)
- Security incident investigation and forensics
- Anomaly detection and threat hunting
- Operational troubleshooting and root cause analysis

This guide explains how to integrate Scaleway Audit Trail with Splunk.

The integration consists of two components, both installable and configurable via Splunk:

- Scaleway Add‑on for Splunk – Collects Audit Trail events from your Scaleway infrastructure and forwards them securely to Splunk Enterprise/Cloud.

> - Scaleway App for Splunk – Visualizes the ingested events with ready‑made dashboards and security analytics; you can also build custom dashboards.

Reviewer: The text for the 2 bullet points above is a tad long so each bullet point spans on 2 lines. Because of this, I'm wondering if we should put the first part of each bullet point in bold. See Slack for team question :eyes:

## Scaleway Add-on for Splunk

### Overview

> The Scaleway Add-On for Splunk acts as a data collection layer that ingests audit logs from Scaleway infrastructure into a Splunk deployment. It connects to Scaleway accounts via API credentials, allowing administrators to collect:

Reviewer 2: ```suggestion
The Scaleway Add-on for Splunk acts as a data collection layer that ingests audit logs from Scaleway infrastructure into a Splunk deployment. It connects to Scaleway accounts via API credentials, allowing administrators to collect:
```

- [Owner](/iam/concepts/#owner) status or [IAM permissions](/iam/concepts/#permission) allowing you to perform actions in the intended Organization
- API credentials (Access Key ID and Secret Key) with Audit Trail (and optionally IAM and Instance) read permissions. For details, see [Configure an API key and permissions](#configure-an-api-key-and-permissions).
- Opened port 443 to allow the Scaleway Add-on to communicate with the Scaleway servers
- Splunk Enterprise or Splunk Cloud Platform (supported versions: 9.4.x, 10.0.x, 10.2.x, 10.4.x)
- The Splunk `admin` role assigned to you

<Message type="note">
Because the Scaleway Add-on operates on the Splunk platform, it inherits the system requirements of your underlying Splunk environment:

- For Splunk Enterprise deployments, see [System Requirements](https://help.splunk.com/en/splunk-enterprise/get-started/install-and-upgrade/10.4/plan-your-splunk-enterprise-installation/system-requirements-for-use-of-splunk-enterprise-on-premises) in the Splunk Enterprise Installation Manual.
- If you are running the add-on entirely within Splunk Cloud, no additional platform requirements apply.

> - If you use on-premise forwarders to route data to Splunk Cloud, see the [System Requirements](https://help.splunk.com/en/splunk-enterprise/get-started/install-and-upgrade/10.4/plan-your-splunk-enterprise-installation/system-requirements-for-use-of-splunk-enterprise-on-premises) in the Splunk Enterprise Installation Manual for forwarder specifications.

Reviewer: We didn't formalized this, but it was my understanding that the "see blablabla" structure was for Scaleway links only, and we are authorized to use other verbs for external links. I might have mixed things up though

Author: Noted but will leave it as-is for now.

</Message>

### Install the add-on

You can install the Scaleway Add-on via Splunk Web or the command line. The add-on is supported across all Splunk Enterprise and Splunk Cloud instance types, including indexers, search heads, and forwarders.

1. Download the Scaleway Add-on from [Splunkbase](https://splunkbase.splunk.com/).
2. Determine the best installation location and method for your deployment architecture.
3. Complete any required prerequisite steps.

> 4. Execute the installation.

Reviewer 2: I'd say "run the Scaleway Add-on installer" but this is okay.

Author: Thanks, good point. Since I was not able to test the installation, I will leave it as-is now. Because it is kind of broader like this and hence more "permissive" as to what happens during installation. But under any other circumstances I would accept your suggestion.

For detailed instructions about how to install the add-on in a single-instance deployment, a distributed deployment, or Splunk Cloud, see the relevant sections in the [Installing add-ons](https://docs.splunk.com/Documentation/AddOns/released/Overview/Installingadd-ons) Splunk documentation:

- [Single-instance Splunk Enterprise](https://docs.splunk.com/Documentation/AddOns/released/Overview/Singleserverinstall)
- [Distributed Splunk Enterprise](https://docs.splunk.com/Documentation/AddOns/released/Overview/Distributedinstall)
- [Splunk Cloud](https://docs.splunk.com/Documentation/AddOns/released/Overview/SplunkCloudinstall)

For distributed environments, the next section provides additional information.

> #### Installation in distributed environments

Reviewer 2: This is typically the type of info that could be collapse by default. So we could use <Accordion> here to have a shorter page.

| Universal forwarders   | No        | No       | Data collection via universal forwarders is unsupported. <br/>Because this add‑on utilizes modular inputs that depend on Python and the Splunk REST handler, a heavy forwarder is required. |

The following table provides a quick reference for the compatibility of the add-on with Splunk distributed deployment features.

| Distributed deployment feature | Supported | Comments |
|--------------------------------|-----------|-----------|
| Search head clusters           | Yes       | Disable add‑on visibility on search heads. |
| Indexer clusters               | Yes       | This method is supported only for deploying the unconfigured add‑on. All configuration must be completed via the user interface of the add‑on on a single, designated node. |
| Deployment server              | Yes       | This method is supported only for deploying the unconfigured add‑on. All configuration must be completed via the user interface of the add‑on on a single, designated node. |

### Configure an API key and permissions

> To start collecting data from the Scaleway Audit Trail, Identity and Access Management (IAM), and Compute APIs using the add-on, you must generate an API key in the Scaleway console and assign the appropriate permissions.

Reviewer 2: There are 3 "and" in the sentence. Using IAM directly removes one and is ok at this point of the page.
```suggestion
To start collecting data from the Scaleway Audit Trail, IAM, and Compute APIs using the add-on, you must generate an API key in the Scaleway console and assign the appropriate permissions.
```

#### Configure an input

A Splunk input is a data ingestion source that brings raw data into the Splunk platform.

To complete the setup of the add-on, you must configure data inputs on the specific Splunk node handling data collection for the add-on (usually a heavy forwarder).
Scaleway recommends using Splunk Web for this configuration, but it is also possible to [configure inputs manually via a configuration file](#appendix-manual-input-configuration).

Once the inputs are configured, the Scaleway Add-on continuously polls Scaleway Audit Trail APIs at the specified intervals and forwards events to Splunk for indexing and analysis.

To retrieve Audit Trail events/IAM logs/Instance server events for a Scaleway Organization and/or Project, complete the following steps.

1. On your Splunk platform instance, go to the Splunk Web homepage, then select **Scaleway Add-on** in the navigation.
2. Go to the **Inputs** tab and click the **Create New Input** button.
        <Lightbox image={image2} alt="A screenshot showing the Create New Input button and drop-down list in Splunk" size="large" />

> 3. Select the relevant input type from the drop-down list.

Reviewer: Brand new guidelines would say "from the dropdown" ✨

>     | Input type                        | Scaleway endpoint that will be queried                 |

Reviewer: Should we avoid the future tense here, or is it absolutely necessary? Does "Queried Scaleway endpoint" works?

    | Audit Trail authentication events | [/audit-trail/v1alpha1/regions/\{region\}/authentication-events](https://www.scaleway.com/en/developers/api/audit-trail/authentication-events) |
    | Audit Trail system events         | [/audit-trail/v1alpha1/regions/\{region\}/system-events](https://www.scaleway.com/en/developers/api/audit-trail/system-events) |
    | IAM logs                          | [/iam/v1alpha1/logs](https://www.scaleway.com/en/developers/api/iam/logs) |
    | IAM users                         | [/iam/v1alpha1/users](https://www.scaleway.com/en/developers/api/iam/users) |
    | IAM groups                        | [/iam/v1alpha1/groups](https://www.scaleway.com/en/developers/api/iam/groups) |
    | IAM applications                  | [/iam/v1alpha1/applications](https://www.scaleway.com/en/developers/api/iam/applications) |
    | Instance servers                  | [/instance/v1/zones/\{zone\}/servers](https://www.scaleway.com/en/developers/api/instance/v1/instances) |

    A window appears prompting you to enter details for your input.

> 4. Provide the required information (e.g., polling interval, Scaleway region). Obligatory fields  are marked with an asterisk.

Reviewer: ```suggestion
4. Provide the required information (e.g., polling interval, Scaleway region). Obligatory fields are marked with an asterisk.
```

    Note that:
    - The minimum allowed polling **Interval** is 1 poll every 60 seconds.
    - The **Index** must correspond to the index in which the Audit Trail events data should be stored. The default is `main`.
    - The **Project ID** field is optional. If no Project ID is given, the input will retrieve data from all the Scaleway Projects. 
        
        To retrieve your Project ID, go to the [Scaleway console](https://account.scaleway.com/), select **Project Dashboard** in the side menu, then click **Copy ID**.

>     - The **Product name** field is optional. If no Product name is given, the input will retrieve data from all the Scaleway products. 

Reviewer: ```suggestion
    - The **Product name** field is optional. If no product name is given, the input will retrieve data from all the Scaleway products. 
```

Reviewer: (ou alors il faut mettre une majuscule dans la ligne suivante ^^)

>         To retrieve a product name, go to the [Scaleway console](https://account.scaleway.com/), select **Monitoring > Audit Trail** in the side menu. The Audit Trail dashboard opens. 

Reviewer: ```suggestion
        To retrieve a product name, go to the [Scaleway console](https://account.scaleway.com/), select **Monitoring** > **Audit Trail** in the side menu. The Audit Trail dashboard opens. 
```

>         On the **Events** tab, select the **Product** field. A drop-down list appears. Use the product name as displayed in the drop-down list. However, make sure that you use lowercase letters only (for example, `instance`) and omit the copyright sign from the name. {/* EDITORIAL COMMENT: Is this correct? */}

Reviewer: The whole section about retrieving product names is written like a numbered steps component but is not a numbered steps component. I would either turn it into a. b. c. steps or simplify the content for a conversational approach.

5. Verify that data is successfully arriving by running the following search on your search head.

    | Input type                        | Search to run                                          |
    | --------------------------------- | ------------------------------------------------------ |
    | Audit Trail events                | `sourcetype=scaleway:audittrail_events*`               |
    | Audit Trail authentication events | `sourcetype=scaleway:audittrail_authentication_events` |
    | Audit Trail system events         | `sourcetype=scaleway:audittrail_system_events`         |
    | IAM logs                          | `sourcetype=scaleway:iam:logs`                         |
    | IAM users                         | `sourcetype=scaleway:iam:users`                        |
    | IAM groups                        | `sourcetype=scaleway:iam:groups`                       |
    | IAM applications                  | `sourcetype=scaleway:iam:applications`                 |
    | Instance servers                  | `sourcetype=scaleway:instance`                         |

>         If you do not see any events, check the **Troubleshooting** tab on your data collection node to verify that your accounts, forwarders, and inputs are all configured successfully.

Reviewer: "configured successfully" sounds weird to me, because the configuration is something you take active part into (to my understanding) while the success is the result following an action. I might lack context and understanding about the configuration, if some elements were "automatic" somewhere your wording is fine, but if not, I'd use another approach (like "verify that your accounts, forwarders, and inputs are all properly configured").

#### Configure optional proxy and log level settings

You can configure optional proxy and log level settings on the Splunk platform instance that you have designated as your configuration server for the add-on.

The Scaleway Add-on supports HTTP/SOCKS proxies.

1. On your Splunk platform instance, go to the Splunk Web homepage, then select **Scaleway Add-on** in the navigation. The **Configuration** tab of the add-on opens.
2. If you need to use a proxy: 
    1. Click the **Proxy** tab.
    2. Check the **Enable** box.
    3. Add your proxy details. If your proxy server does not require authentication, leave the **Username** and **Password** fields empty. 
    4. Click **Save**.
5. If you want to change the logging level:
    1. Click the **Logging** tab.
    2. Select the **Log Level**.
    3. Click **Save**.

### Upgrade the add-on

> To upgrade the app, complete the following steps.

Reviewer 2: ```suggestion
To upgrade the add-on, complete the following steps.
```
The whole sentence could also be removed. In does not actually appear in the Upgrade the app section, and it is ok I guess.

1. Disable all inputs.
2. Download the latest version of the Scaleway Add-on from [Splunkbase](https://splunkbase.splunk.com/).
3. [Install the add-on](#install-the-add-on) across your deployment.
4. Enable all the configured inputs to resume the data collection.

### Troubleshoot the add-on

For details, see [I have issues with my Splunk integration](/audit-trail/troubleshooting/issues-with-splunk-integration).

## Scaleway App for Splunk

> ### Overview

Reviewer: I've hesitated underlying the first "overview" directly after a h2 ; I think this is something we need to discuss for the best practices because except from a very very clean navigation structure it doesn't bring much; I kind of find it natural to have introductory content below a title before diving into things.

Now that there are identicals ### Overview titles in the document, I'm worried about: 
- The research behavior
- The number of items in the right-hand side navigation menu

Don't hesitate to discuss this with Firdevs as well if this is something we are unsure about.

The Scaleway App for Splunk is a visualization layer built for Scaleway Audit Trail data. While the add-on handles raw data collection, the application provides pre-built dashboards for security monitoring, with the option to create custom dashboards. It also records user access patterns and maintains chronological logs of resource changes to assist with event correlation. Built-in alerting logic notifies administrators about dropped events or security anomalies.

### Before you start

To complete the actions presented below, you must have:

- Splunk Enterprise or Splunk Cloud Platform (supported versions: 9.4.x, 10.0.x, 10.2.x, 10.4.x)
- The Splunk `admin` role assigned to you for installing the app
- At least the Splunk `power` role assigned to you for configuring the app

> - Installed and configured Scaleway Add-On for Splunk

Reviewer 2: ```suggestion
- Installed and configured Scaleway Add-on for Splunk
```

- Audit Trail data flowing into Splunk

<Message type="note">
Because the Scaleway App operates on the Splunk platform, it inherits the system requirements of your underlying Splunk environment:

- For Splunk Enterprise deployments, see [System Requirements](https://help.splunk.com/en/splunk-enterprise/get-started/install-and-upgrade/10.4/plan-your-splunk-enterprise-installation/system-requirements-for-use-of-splunk-enterprise-on-premises) in the Splunk Enterprise Installation Manual.
- If you are running the app entirely within Splunk Cloud, no additional platform requirements apply.
</Message>

### Install the app

You can install the Scaleway App via Splunk Web or the command line. Note that the app is supported only on search head instances.

1. Download the Scaleway App from [Splunkbase](https://splunkbase.splunk.com/).
2. Determine the best installation method for your deployment architecture.
3. Complete any required prerequisite steps.

> 4. Execute the installation.

Reviewer 2: Same as above. I'd say "run the Scaleway App installer" but this is okay.

Author: Same comment as above.

> For detailed instructions about how to install the app in a single-instance deployment, a distributed deployment, or Splunk Cloud, see the relevant sections in the [Installing add-ons](https://docs.splunk.com/Documentation/AddOns/released/Overview/Installingadd-ons) Splunk documentation:

Reviewer 2: Does this link really applies to apps as well ?

Author: Great catch. I double-checked and this is the info I got. I will go back to the person who provided the link and ask for a confirmation.

Update --> I got the confirmation: saying "add-on" is OK.

        1. Go to `$SPLUNK_HOME/etc/shcluster/apps/DA-scaleway_app/local/macros.conf`.
        2. Add the following stanza and replce `&lt;value&gt;` with the correct index.

            ```bash
            [scaleway_index]
            definition = index=&lt;value&gt;
            ```

        3. Save the file.
        4. To apply the Knowledge Bundle across all Search Head nodes, follow the instructions in [Use the deployer to distribute apps and configuration updates](https://help.splunk.com/en/splunk-enterprise/administer/distributed-search/10.4/update-search-head-cluster-members/use-the-deployer-to-distribute-apps-and-configuration-updates) in the Splunk documentation.
    </TabsTab>
</Tabs>

#### Configure and enable alerts

The Scaleway App allows you to configure alerts to monitor:

- Spikes in error rates compared to the previous day by enabling the "Scaleway - Increased number of errors from yesterday" alert.
- Increases in failed login attempts by enabling the "Scaleway - Increased number of failed logins" alert.
- Interruptions in data ingestion by enabling the "Scaleway - No data ingesting in the last hour" alert.

> **Prerequisites:** 

Reviewer: I'm not too sure about using bold in the subsections here, because in the preview, bold seems to have a higher weight than h4 content; this could be misleading. However, I understand the need for hierarchy and if you don't have any other satisfactory solution, I'll talk with the website team to see what we can adjust in terms of headings. Don't feel obligated to change!
